test/Tests/src/Data/Text_Spec.enso

```diff
@@ -627,6 +638,72 @@ spec =
             "foobar" . ends_with "" Regex_Matcher.new . should_be_true
             "" . ends_with "" Regex_Matcher.new . should_be_true
 
+        Test.specify "should allow to pad a text" <|
+            "Hello World!".pad 15 . should_equal "Hello World!   "
+            "HELLO".pad 9 "AB" . should_equal "HELLOABAB"
+            "HELLO".pad 8 "AB" . should_equal "HELLOABA"
+            "HELLO".pad 8 "AB" Location.Start . should_equal "BABHELLO"
+            "".pad 4 . should_equal "    "
+            "A".pad 3 "" . should_fail_with Illegal_Argument_Error
+            "ABCDE".pad 3 "" . should_fail_with Illegal_Argument_Error
+            "".pad 0 "" . should_fail_with Illegal_Argument_Error
+
+            "".pad 0 . should_equal ""
+            "ABC".pad 3 . should_equal "ABC"
+            "AB".pad -1 . should_equal "AB"
+            "ABC".pad -100 . should_equal "ABC"
+
+            'a\u{301}'.pad 2 . should_equal 'a\u{301} '
+            "".pad 2 'a\u{302}' . should_equal 'a\u{302}a\u{302}'
+            'XX'.pad 5 'yy\u{301}' . should_equal 'XXyy\u{301}y'
```

Comment:
can we do one more test case where the pad_string is truncated on a mulit-char graphene cluster please